doc/developer/server_side.rst.mako

```diff
@@ -9,107 +9,100 @@ Create development environment in a project
+code, and a make environment for it. So do that first, as described below.
 
-  You can now check out your development branch if necessary.
+Install c2cgeoportal from source
+................................
 
-* Edit your ``<user>.mk`` to have something like this:
+Check out c2cgeoportal from GitHub:
 
-  .. code:: make
+.. prompt:: bash
 
-    DEVELOPMENT = TRUE
-    REQUIREMENTS = -e ../c2cgeoportal
+    git clone git@github.com:camptocamp/c2cgeoportal.git
+    cd c2cgeoportal
 
-    include <package>.mk
+Pull all the base images:
 
-* Uninstall the regular c2cgeoportal egg from the virtual environment:
+.. prompt:: bash
 
-  .. prompt:: bash
+    ./docker-run make pull
 
-    ./docker-run pip uninstall c2cgeoportal
+If you modify something in the folder ``docker/build`` build the base image:
 
-* Remove/comment the following line in the CONST_requirements.txt file:
+.. prompt:: bash
 
-  .. code:: make
+    docker build --tag camptocamp/geomapfish-build-dev:${major_version} docker/build
 
-    c2cgeoportal==1.6.0
+Build everything:
 
-* Build your application application:
+.. prompt:: bash
 
-  .. prompt:: bash
+    ./docker-run make build
 
-    ./docker-run rm /build/requirements.timestamp && ./docker-run make build
+c2cgeoportal has two types of tests: unit tests and functional tests. The unit
+tests are self-contained, and do not require any specific setup. The functional
+tests require to run with ``docker-compose-run``.
 
-.. note:: Print performance issue
+docker-run will:
 
-   When restarting the print server frequently, performance issues may randomly be observed.
-   This is done in random number generation.
+* Run the given command in the docker build image.
+* Mount the current directory in /src.
+* Use the current user and home directory.
+* Use a named volume related to the current folder in ``/build``.
 
-   To improve the performances you should add in the ``/srv/tomcat/tomcat1/bin/setenv-local.sh`` file:
+docker-compose will also:
 
-   .. code:: bash
+* Have a testing database and a testing MapServer.
 
-      export ADD_JAVA_OPTS="-Djava.security.egd=file:/dev/./urandom"
+Docker images dependencies::
 
+.. image:: docker.png
+.. source file is docker.dia.
 
-Tests
------
+Image nomoclature:
 
-Running tests
-~~~~~~~~~~~~~
-
-To be able to run c2cgeoportal tests you need to have the c2cgeoportal source
-code, and a make environment for it. So do that first, as described below.
+* ``camptocamp/geomapfish-*`` for all images generated by/for c2cgeoportal
+* ``camptocamp/geomapfish-test-*`` for all images used only for the CI (not pushed on a Docker repository)
+* ``camptocamp/testgeomapfish-*`` the images related to the generated application,
+        will be ``camptocamp/<package>-*`` in the client application.
 
-Install c2cgeportal from source
-...............................
+Unit tests
+..........
 
-Check out c2cgeoportal from GitHub:
+To run the unit tests do this, buils all the dependencies:
```

Comment:
buil**d**s

"do this" is not useful for me.

I would prefer : "Before to run the tests, install and build all dependencies :"

Reply:
+1, thanks :-)